tests/test_metrics.py

```diff
@@ -178,7 +178,10 @@ def test_cuda_available_model_cpu(self, model, is_available_mock):
         is_available_mock.return_value = True
         model.device = 'cpu'
 
-        with pytest.warns(UserWarning):
+        with pytest.warns(
+            UserWarning,
+            match='CUDA available but model device is set to CPU - is this desired?'
```

Comment:
Does this also need a `re.escape`, since `?` is a regex-specific character?

Reply:
Good call!